react/src/lib/components/WebvizDialog/WebvizDialog.tsx

```diff
@@ -0,0 +1,732 @@
+        if (!props.disableEscapeKeyDown) {
+            window.addEventListener("keyup", handleEscapeKeyUp);
+        }
+
+        return () => {
+            if (dialogTitleRef.current) {
+                dialogTitleRef.current.removeEventListener(
+                    "mousedown",
+                    handleMouseDown
+                );
+                dialogTitleRef.current.removeEventListener(
+                    "touchstart",
+                    handleTouchStart
+                );
+            }
+
+            window.removeEventListener("blur", handleBlur);
+            document.removeEventListener("mousemove", handleMouseMove);
+            document.removeEventListener("mouseup", handleMouseUpAndTouchEnd);
+            document.removeEventListener("touchmove", handleTouchMove);
+            document.removeEventListener("touchend", handleMouseUpAndTouchEnd);
+
+            if (!props.disableEscapeKeyDown) {
+                window.removeEventListener("keyup", handleEscapeKeyUp);
+            }
+        };
+    }, [
+        dialogRef.current,
+        dialogTitleRef.current,
+        props.disableEscapeKeyDown,
+        props.disableDraggable,
+    ]);
+
+    React.useEffect(() => {
+        if (!dialogRef.current) {
+            return;
+        }
+
+        let previousWindowWidth = window.innerWidth;
+        let previousWindowHeight = window.innerHeight;
+
+        const handleWindowWidthResize = () => {
+            if (!open || !dialogRef.current || initialDialogWidth === null) {
+                return;
+            }
+
+            const dialogLeft = dialogRef.current.getBoundingClientRect().left;
+            const dialogWidth = dialogRef.current.getBoundingClientRect().width;
+            const windowWidth = window.innerWidth;
+            const deltaWidth = windowWidth - previousWindowWidth;
+
+            const wasDialogOutsideWindow =
+                dialogLeft < 0 ||
+                dialogLeft + dialogWidth > previousWindowWidth;
+
+            // Width resize: Handle width and left position
+            if (wasDialogOutsideWindow) {
+                if (dialogLeft < 0) {
+                    const newDialogWidth = Math.min(
+                        Math.max(props.minWidth || 0, dialogWidth + deltaWidth),
+                        initialDialogWidth
+                    );
+                    if (
+                        deltaWidth < 0 &&
+                        windowWidth < dialogLeft + dialogWidth + margin.right
+                    ) {
+                        // Decrease dialog width when window width decrease
+                        setDialogWidth(newDialogWidth);
+                    } else if (
+                        deltaWidth > 0 &&
+                        windowWidth > dialogLeft + dialogWidth + margin.right
+                    ) {
+                        // Increase dialog width when window width increase
+                        setDialogWidth(newDialogWidth);
+                    }
```

Comment:
Combine these two checks by creating a variable for `dialogLeft + dialogWidth + margin.right`?